converters/toZigbee.js

```diff
@@ -1763,8 +1763,16 @@ const converters = {
                 return {state: {[`operation_mode${button !== 'single' ? `_${button}` : ''}`]: value.state}};
             } else if (['QBKG25LM', 'QBKG26LM', 'QBKG39LM'].includes(meta.mapped.model)) {
                 const lookupState = {control_relay: 0x01, decoupled: 0x00};
-                await entity.write('aqaraOpple', {0x0200: {value: lookupState[value.state], type: 0x20}}, manufacturerOptions.xiaomi);
-                return {state: {operation_mode: value.state}};
+                // Support existing syntax of a nested object just for the state field. Though it's quite silly IMO.
+                const targetValue = value.hasOwnProperty('state') ? value.state : value;
+                await entity.write('aqaraOpple', {0x0200: {value: lookupState[targetValue], type: 0x20}}, manufacturerOptions.xiaomi);
+                if (meta.mapped.meta.multiEndpoint) {
```

Comment:
I don't believe this `if` branch is needed. Zigbee2MQTT will automatically add the endpoint to all properties when setting via e.g. `zigbee2mqtt/my_ QBKG26LM/left/set`

Reply:
Which is wierd, because that's not what I observed was happening with the original code. 
It resulted in a state of operation_mode instead of say, operation_mode_left

Reply:
@Koenkk I double checked against the on_off converters, it does more or less the same thing there, so yeah. Changed the code to use the helper function from util to match.

Reply:
That is indeed weird, I can't remembers why https://github.com/Koenkk/zigbee2mqtt/blob/9bd46932c365e886ed0b1f9b8740d8cce632230c/lib/extension/publish.js#L232 is needed, can you try removing it?

Reply:
Introduced here. https://github.com/Koenkk/zigbee2mqtt/commit/a77e8578688a394a3bd34fd179b983f42204a219

Reply:
Though personally, I don't think the inconsistency in handling that in z2m / herdsman-converters is within the scope of this PR though.

Reply:
Made this consistent now in the latest dev, this should be handled in z2m and not in herdsman converters: https://github.com/Koenkk/zigbee2mqtt/commit/1985e2451797adbd91496012e869e151aa0fe7f6

Reply:
In which case, now you'd need to go through all the tz converters and change those that explicitly return a suffixed endpoint... or ignore properties that are already suffixed. tz.on_off for example, returns it suffixed already, that change will now result in... double suffixing?

Also, what about the fz path which is still suffix-ed manually? Because receive.js sure as heck doesn't handle that.

Reply:
Let's keep the discussion of https://github.com/Koenkk/zigbee2mqtt/commit/1985e2451797adbd91496012e869e151aa0fe7f6#r52979633 here.

I'm wondering if we can make this converter more consistent:
- QBKG24LM: `zigbee2mqtt/FRIENDLY_NAME/system/set` -> `zigbee2mqtt/FRIENDLY_NAME/set` should also work (since it will use the first endpoint). Do you have this device to test with?
- QBKG25LM: we can select the correct endpoint in the converter based on the `button`.

Reply:
For QBKG25LM, QBKG26LM, QBKG39LM, these run the same code path where it sends to aqaraOpple cluster on the same endpoint as the onOff clusters. At least for what I can see in the converter (I only have the 26), so that's not a problem.

Maybe if it can detect what button from the *endpoint* and then just send to the 1st endpoint instead, that should make the converter's function overall more consistent between models, hiding that implementation detail from the user.

(Also, I have no idea how exposes will work for this wierd setup)

Reply:
Thanks, reviewed the PR and looks good to go, can I merge it?

Regarding docs, I will remove all these sections: https://www.zigbee2mqtt.io/devices/QBKG04LM.html#decoupled-mode, since you've added them the `expose` now (great!) docs will be automatically generated under https://www.zigbee2mqtt.io/devices/QBKG04LM.html#exposes

Reply:
Yeah, go ahead. So far I've been running it on my local installation and it appears to be working fine.

Reply:
Big thanks for cleaning this up :)